control/pid_longitudinal_controller/src/longitudinal_controller_utils.cpp

```diff
@@ -129,11 +129,16 @@ double calcElevationAngle(const TrajectoryPoint & p_from, const TrajectoryPoint
 }
 
 Pose calcPoseAfterTimeDelay(
-  const Pose & current_pose, const double delay_time, const double current_vel)
+  const Pose & current_pose, const double delay_time, const double current_vel,
+  const double current_acc)
 {
+  if (delay_time <= 0.0 || current_vel <= 0.0) {
+    return current_pose;
+  }
```

Comment:
I think delay compensation should work when the vehicle drives backward as well.

Reply:
Yes right, running_distance should not be negative so we can use abs(current_vel), what do you think?

Reply:
With the latest commit, this was also changed.